aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -1498,6 +1569,11 @@ module aptos_framework::delegation_pool {
                 pool.observed_lockup_cycle,
                 pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR)
             );
+
+            // update the commission percentage for the new lockup cycle
+            if (exists<CommissionPercentage>(pool_address)) {
```

Comment:
Let's say that the commission has been updated during lockup cycle _N_ on the stake pool. After lockup cycle _N+1_ starts and until the 1st `synchronize_delegation_pool`, rewards are produced and will be shared with the operator using the old commission fee.
Additionally, the new commission fee takes effect when a new _observed_ lockup cycle is detected. not an actual lockup cycle (within the stake pool). To address these issues:

- owner calls `update_commission_percentage`
- owner `unlock`s some small amount of APT and waits for it to be inactivated - this ensures the observed lockup cycle advances together with the actual lockup cycle
- during the first epoch of the new lockup, owner calls `synchronize_delegation_pool` to ensure that all rewards produced starting this new lockup cycle are shared using the new commission

Reply:
Ensuring that the new commission is applied exactly when the next lockup starts cannot be achieved at the contract level because: rewards are produced continuously by the stake pool and summed up, but there is no callback that allows to snapshot state at the beginning of the lockup cycle.

Reply:
However, a snapshotting mechanism could be implemented at the stake pool level when the lockup cycle is renewed.
This would include the stake amounts on the validator at that time (or any other state that may be relevant to other modules). Unfortunately, multiple past snapshots should be stored, not just the last one, as the delegation pool might be idle for more than 1 lockup period.
This doesn't affect the integrity of the staking module, but it is not consistent with its overall implementation.

Reply:
> owner unlocks some small amount of APT and waits for it to be inactivated - this ensures the observed lockup cycle advances together with the actual lockup cycle

As the alternative that we discussed offline, we are using `effective_after_secs` (
https://github.com/aptos-labs/aptos-core/pull/10226/files#diff-072b3b0506ff0035c65fe82ead317e994c034d27f31ce21414d31a6675a7c40aR325)